### PR TITLE
Add Location Functionality on Map View

### PR DESCRIPTION
**Pull request by leeminkyu16**

### CODE REVIEW

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt [hunk 1]
```diff
@@ -87,15 +94,53 @@ class HomePageViewModel @Inject constructor(
...
+		.map { getListingParams ->
+			if (getListingParams.filters.addressSearch != null) {
+				val splitAddress = getListingParams.filters.addressSearch.split(",").toTypedArray()
+				runCatching {
+					runBlocking {
+						geocodeApi.forwardGeocode(
+							addressLine = if (splitAddress.isNotEmpty()) splitAddress[0] else "",
+							addressCity = if (splitAddress.size >= 2) splitAddress[1] else "",
+							addressPostalcode = if (splitAddress.size >= 3) splitAddress[2] else "",
+							addressCountry = if (splitAddress.size >= 4) splitAddress[3] else "Canada",
```
Comment (henryhu1) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt:109]: I've seen this somewhere before... 🤔 (not important)

**Review by henryhu1 — APPROVED** [review]
*I* think it's pretty

File: android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt (hunk 1 above)
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt:109]: That's crazy. It's like I copied it from somewhere lol

**Comment by leeminkyu16** [comment]
I think you're pretty Henry <3

**Comment by henryhu1** [comment]
> I think you're pretty Henry <3

😳 :hand_with_index_finger_and_thumb_crossed:

**Review by abhaykshar — APPROVED** [review]
huge

**Pull request merged**